.github/workflows/reusable-release.yml

```diff
@@ -110,12 +110,31 @@ jobs:
     name: 🎊 Test & Version
     needs: init
     runs-on: ubuntu-latest
+    outputs:
+      library_build_major_java_version: ${{ steps.establish_java_for_library_build.outputs.library_build_major_java_version }}
     steps:
       - uses: actions/checkout@v4
+      - id: establish_java_for_library_build
+        name: Establish library build Java version
+        run: |
+          if [ ! -f .tool-versions ]; then
+            echo "::error title=Missing .tool-versions file::gha-scala-library-release-workflow requires an asdf-format .tool-versions file to establish the Java version for the build."
+            exit 1
+          fi
+          LIBRARY_BUILD_MAJOR_JAVA_VERSION=$( grep -Eo 'java [[:alnum:]-]+-[[:digit:]]+' .tool-versions | rev | cut -d'-' -f1 | rev )
```

Comment:
This regex has been manually checked by running it on the output of `asdf list-all java` (1238 versions), ie:

```
asdf list-all java | sed -e 's/^/java /' | grep -Eo 'java [[:alnum:]-]+-[[:digit:]]+' | rev | cut -d'-' -f1 | rev > major-java-versions.csv
```

The spreadsheet of the results is [here](https://docs.google.com/spreadsheets/d/1Gg_JHjBAQv-16KzghNGwbmIjTG6iH_eOnuWUtadJlyg/edit?usp=sharing) - a quick skim through doesn't show any obvious problems to me.